cmd/target-browser/cmd/get.go

```diff
@@ -0,0 +1,85 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	targetbrowser "github.com/trilioData/tvk-plugins/tools/target-browser"
+)
+
+var targetBrowserAuthConfig = &targetbrowser.AuthInfo{}
+
+// getCmd represents the get command
+var getCmd = &cobra.Command{
+	Use:   "get",
+	Short: "get command retrieves specific resource",
+	Long: `Gets specific resource[backup, backupPlan, metadata, etc] which retrieves single
+object or list of objects of that resource.
+`,
+	Example: `  # List of backups	
+  kubectl target-browser get backup
+
+  # List of backupPlans
+  kubectl target-browser get backupPlan
+
+  # Metadata of specific backup object
+  kubectl target-browser get metadata
+
+  # Specific backup
+  kubectl target-browser get backup --backup-uid <uid> --backupplan-uid <uid>
+`,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+		err = validateInput(cmd)
+		if err != nil {
+			return err
+		}
+
+		if targetBrowserAuthConfig, err = TargetBrowser.Authenticate(context.Background()); err != nil {
+			return err
+		}
+		// TODO: remove this line and use 'targetBrowserAuthConfig' as a receiver for getCmd's child commands
+		fmt.Println(targetBrowserAuthConfig)
+		return nil
+	},
+}
+
+func init() {
+	getCmd.PersistentFlags().IntVar(&TargetBrowser.Pages, pagesFlag, pagesDefault, pagesUsage)
+	getCmd.PersistentFlags().IntVar(&TargetBrowser.PageSize, pageSizeFlag, pageSizeDefault, pageSizeUsage)
+	getCmd.PersistentFlags().StringVar(&TargetBrowser.OrderBy, orderByFlag, orderByDefault, orderByUsage)
+
+	rootCmd.AddCommand(getCmd)
+}
+
+func validateInput(cmd *cobra.Command) error {
```

Comment:
You have to validate input manually? Is there a way cobra can handle this?

Reply:
These are our custom validations, no package can give us these validations which can used readily. Only thing, that can be managed or cobra can provide is where should this func be accommodated and called from. I thought, `cmd.PersistentPreRunE` was best place to have this validation func so put it there.

Reply:
basic and primitive validations cobra itself handles, we don't need to call any func for that.